From 91c9f534f690cf815e9766727a99c9da772f49ed Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Sun, 15 Jun 2014 19:57:19 -0700
Subject: [PATCH] Adapting linker arguments.

 - Adding -L<prefix>/lib to default linker arguments.
 - On Linux, and -rpath.
---
 lib/Driver/Tools.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index e49267a..34e28f7 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -210,6 +210,14 @@ static void AddLinkerInputs(const ToolChain &TC,
   //                and only supported on native toolchains.
   if (!TC.isCrossCompiling())
     addDirectoryList(Args, CmdArgs, "-L", "LIBRARY_PATH");
+
+  if (!TC.isCrossCompiling()) {
+    CmdArgs.push_back(Args.MakeArgString("-L" + D.InstalledDir + "/../lib"));
+
+    if (TC.getTriple().getOS() == llvm::Triple::Linux)
+      CmdArgs.push_back(Args.MakeArgString("-rpath=" + D.InstalledDir + "/../lib"));
+  }
+
 }
 
 /// \brief Determine whether Objective-C automated reference counting is
-- 
1.9.3

